multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v2/ResourceLiveParameterResolver.java

```diff
@@ -0,0 +1,149 @@
+                replaceValueInResourceParameter(module, resourceParameterValue, resourceParameter, ref,
+                                                castToString(requiredDependencyValue));
+            }
+        }
+    }
+
+    private boolean isStringValueReplaceable(Object object) {
+        return object instanceof String && idleToLiveParameterPairs.keySet()
+                                                                   .stream()
+                                                                   .anyMatch(defaultParamPlaceholder -> castToString(object).contains(defaultParamPlaceholder));
+    }
+
+    private void replaceValueInResourceParameter(Module module, String resourceParameterValue, Map.Entry<String, Object> resourceParameter,
+                                                 Reference reference, String requiredDependencyValueString) {
+        Map.Entry<String, String> requiredIdleToLiveParameterPair = getMapEntryOnCondition(requiredDependencyValueString);
+        requiredDependencyValueString = requiredDependencyValueString.replace(requiredIdleToLiveParameterPair.getKey(),
+                                                                              castToString(module.getParameters()
+                                                                                                 .get(requiredIdleToLiveParameterPair.getValue())));
+        requiredDependencyValueString = resourceParameterValue.replace(reference.getMatchedPattern(), requiredDependencyValueString);
+
+        resourceParameter.setValue(requiredDependencyValueString);
+    }
+
+    private void resolveModuleProvidedDependency(Map<String, Object> requiredDependencyMap, Module module) {
+        requiredDependencyMap.entrySet()
+                             .forEach(entry -> updateModuleDependencyFromMap(module, entry));
+    }
+
+    private void updateModuleDependencyFromMap(Module module, Map.Entry<String, Object> entry) {
+        if (isStringValueReplaceable(entry.getValue())) {
+            var idleToLiveParameterPair = getMapEntryOnCondition(castToString(entry.getValue()));
+            entry.setValue(castToString(entry.getValue()).replace(idleToLiveParameterPair.getKey(), castToString(module.getParameters()
+                                                                                                                       .get(idleToLiveParameterPair.getValue()))));
+        } else if (entry.getValue() instanceof Map) {
+            resolveModuleProvidedDependency(castToMap(entry.getValue()), module);
+        }
+    }
+
+    private Map.Entry<String, String> getMapEntryOnCondition(String replaceableString) {
+        return idleToLiveParameterPairs.entrySet()
+                                       .stream()
+                                       .filter(idleToLiveParameterPair -> replaceableString.contains(idleToLiveParameterPair.getKey()))
+                                       .findFirst()
```

Comment:
Not a style comment: What would happen if a parameter references 2 placeholders at once? e.g. '${default-host}-something-${default-uri}'; Not sure if this is even a real case but still

Reply:
they are resolved correctly